### PR TITLE
Add API documentation for the perturbed optimizers.

### PR DESCRIPTION
**Pull request by fabianp**

### CODE REVIEW

File: docs/api.rst [hunk 1]
```diff
@@ -129,6 +129,20 @@ Line search
 
     jaxopt.BacktrackingLineSearch
 
+
+
+Perturbed Optimizers
```
[the remark this answers is not in the document]
Comment (mblondel) [reply docs/api.rst:134]: (for consistency with how we capitalized the other sections)

**Review by mblondel — APPROVED** [review]
LGTM, apart from minor suggestion below

File: docs/api.rst (hunk 1 above)
[the remark this answers is not in the document]
Comment (fabianp) [reply docs/api.rst]: indeed, thanks! fixed that

**Review by q-berthet — APPROVED** [review]
Thanks!

**Pull request merged**